### ISSUE TITLE
[Local modal]: Able to navigate outside of modal – React Safari Mobile

### ISSUE THREAD

**Issue by oliviaflory** [carbon-design-system/carbon-for-ibm-dotcom#8560]
### Description

I am able to navigate outside of the modal using VO

Expect the user to be trapped inside modal. 

Tested on [dotcom shell ](https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/react/?path=/story/components-dotcom-shell--micro-footer)

### Component(s) impacted

Locale modal

Expressive modal?

### Browser

Safari

### Carbon for IBM.com version

v1.32.0 rc1

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

carb4IBM.com

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/react/?path=/story/components-dotcom-shell--micro-footer

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by annawen1** [comment]
This is an issue present in the the `ComposeModal` which we are using for `Locale Modal`, I think this is something that should be fixed upstream in Carbon Core

https://react.carbondesignsystem.com/?path=/story/components-composedmodal--default

**Comment by oliviaflory** [comment]
@annawen1 thanks for the update, do you mind opening an issue on the Carbon github and posting the issue number here to track it?
